maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -847,12 +847,33 @@ private DefaultScanResult scanDependencies()
         {
             try
             {
+                DefaultScanResult scanResult = new DefaultScanResult( Collections.EMPTY_LIST );
+                
+                List<String> dependenciesToScanList = new ArrayList( Arrays.asList( getDependenciesToScan() ) );
```

Comment:
No need to create two collections with 
`List<String> dependenciesToScanList = new ArrayList( Arrays.asList( getDependenciesToScan() ) );`
One modifiable collection can be with this code:
`List<String> dependenciesToScanList = new ArrayList();`
`Collections.addAll( dependenciesToScanList, getDependenciesToScan() );`